tests/streamlit_ui_tests/features/merge.spec.ts

```diff
@@ -79,4 +79,144 @@ test.describe('Merge', () => {
         // Check that there are 5 columns still
         await expect(mito.locator('.endo-column-header-container')).toHaveCount(5);
     });
+
+    test('Change Merge Type', async ({ page }) => {
+        const mito = await getMitoFrame(page);
+        await importCSV(page, mito, 'merge.csv');
+        await importCSV(page, mito, 'test.csv');
+        
+        await clickButtonAndAwaitResponse(page, mito, { name: '▾ Merge' })
+        await mito.getByText('Merge (horizontal)').click();
+        await awaitResponse(page);
+    
+        await expect(mito.getByText('Merge Dataframes')).toBeVisible();
+
+        // Wait for the merge to be finished before continuing so adding a column works!
+        await expect(mito.getByText('df_merge')).toBeVisible();
+    
+        // Check that the correct number of rows are present
+        await hasExpectedNumberOfRows(mito, 4);
+
+        const changeMergeType = async (oldMergeType: string, newMergeType: string, expectedRows: number) => {
+            // Change the merge type
+            await mito.locator('.select-text', { hasText: oldMergeType }).click();
+            await mito.locator('.mito-dropdown-item span').getByText(newMergeType, { exact: true }).click();
+            await awaitResponse(page);
+
+            // Check that the correct number of rows are present
+            await hasExpectedNumberOfRows(mito, expectedRows);
+        }
+
+        await changeMergeType('lookup', 'left', 5);
+        await changeMergeType('left', 'right', 5);
+        await changeMergeType('right', 'inner', 4);
+        await changeMergeType('inner', 'outer', 6);
+        await changeMergeType('outer', 'unique in left', 1);
+        await changeMergeType('unique in left', 'unique in right', 1);
+    });
+
+    test('Error handling and changing merge keys / dataframes', async ({ page }) => {
+        const mito = await getMitoFrame(page);
+        await importCSV(page, mito, 'merge.csv');
+        await importCSV(page, mito, 'test.csv');
+        await importCSV(page, mito, 'types.csv');
+        
+        await clickButtonAndAwaitResponse(page, mito, { name: '▾ Merge' })
+        await mito.getByText('Merge (horizontal)').click();
+        await awaitResponse(page);
+
+        // Check that we throw an error because the default merge keys are different types
+        await expect(mito.locator('.text-color-error')).toHaveText('Column1 (object) and Column1 (int64) have different types. Either pick new keys or cast their types.');
+
+        // Test changing the dataframe
+        await mito.locator('.select-text', { hasText: 'types' }).click();
+        await mito.locator('.mito-dropdown-item span').getByText('merge', { exact: true }).click();
```

Comment:
I would love a utility for changing the dataframe as well. Ideally, it would just take an index of the dataframe (either 0 or 1), and the new dataframe name. It also awaits a response as well.